service/src/main/kotlin/fi/espoo/evaka/messaging/daycarydailynote/DaycareDailyNoteController.kt

```diff
@@ -37,16 +48,69 @@ class DaycareDailyNoteController {
         @PathVariable childId: UUID
     ): ResponseEntity<List<DaycareDailyNote>> {
         Audit.DaycareDailyNoteRead.log(user.id)
+
+        acl.getRolesForChild(user, childId)
+            .requireOneOfRoles(UserRole.ADMIN, UserRole.UNIT_SUPERVISOR, UserRole.STAFF, UserRole.MOBILE)
+
         return db.read { it.getChildDaycareDailyNotes(childId) }.let { ResponseEntity.ok(it) }
     }
 
-    @PostMapping("/child")
-    fun createOrUpdateDaycareDailyNoteForChild(
+    @PostMapping("/child/{childId}")
+    fun createDailyNoteForChild(
         db: Database.Connection,
         user: AuthenticatedUser,
+        @PathVariable childId: UUID,
         @RequestBody body: DaycareDailyNote
-    ) {
+    ): ResponseEntity<UUID> {
         Audit.DaycareDailyNoteCreate.log(user.id)
-        return db.transaction { it.upsertDaycareDailyNote(body) }.let { ResponseEntity.ok() }
+
+        acl.getRolesForChild(user, childId)
+            .requireOneOfRoles(UserRole.ADMIN, UserRole.UNIT_SUPERVISOR, UserRole.STAFF, UserRole.MOBILE)
```

Comment:
In order for this to work we need to have the mobile devices showing up in the acl view. Could we add mobile devices to `child_acl_view`?

Reply:
Replacing this check with
```
if (!user.hasOneOfRoles(UserRole.MOBILE)) {
    acl.getRolesForChild(user, childId)
    .requireOneOfRoles(UserRole.ADMIN, UserRole.UNIT_SUPERVISOR, UserRole.STAFF)
}
```
could work?

Reply:
Fixed in https://github.com/espoon-voltti/evaka/pull/582, rebased